Updated pull request

joshzambales opened the pull request:
Changelog: 
-Split app into separate java files
-Added javadocs 
-changed image url to point to iluwatar (UML diagram)
-changed setter injection to constructor injection
-Added intercepting-filter to pom.xml parent as a module
-configured intercepting-filter's pom.xml (my bad, careless mistake)
-separated buttons from button array and changed variable names for more consistent code


Considerations:
To make code more organized and neat, I've decided not to use anonymous classes anymore. Also, if possible, consider retention of FilterManager as I just followed the framework of my source and the delegation is how it manages the FilterChain.

iluwatar commented on README.md:
```diff
@@ -431,6 +437,17 @@ Behavioral patterns are concerned with algorithms and the assignment of responsi
 **Applicability:** Use the Callback pattern when
 * When some arbitrary synchronous or asynchronous action must be performed after execution of some defined activity.
 
+<<<<<<< HEAD
```
Remove this merge marker.

iluwatar commented on README.md:
```diff
@@ -431,6 +437,17 @@ Behavioral patterns are concerned with algorithms and the assignment of responsi
+* a system uses pre-processing or post-processing requests
+* a system should do the authentication/ authorization/ logging or tracking of request and then pass the requests to corresponding handlers 
+* you want a modular approach to configuring pre-processing and post-processing schemes
+=======
```
Remove this merge marker.

iluwatar commented on README.md:
```diff
@@ -452,6 +469,7 @@ Behavioral patterns are concerned with algorithms and the assignment of responsi
 
 **Real world examples:**
 * [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain) prototype inheritance
+>>>>>>> origin/master
```
Remove this merge marker.

iluwatar commented on README.md:
```diff
@@ -497,9 +515,14 @@ The difference is the intent of the patterns. While Proxy controls access to the
 * [Let’s Modify the Objects-First Approach into Design-Patterns-First](http://edu.pecinovsky.cz/papers/2006_ITiCSE_Design_Patterns_First.pdf)
 * [Pattern Languages of Program Design](http://www.amazon.com/Pattern-Languages-Program-Design-Coplien/dp/0201607344/ref=sr_1_1)
 * [Martin Fowler - Event Aggregator](http://martinfowler.com/eaaDev/EventAggregator.html)
+<<<<<<< HEAD
```
Remove this merge marker.

iluwatar commented on README.md:
```diff
@@ -497,9 +515,14 @@ The difference is the intent of the patterns. While Proxy controls access to the
+<<<<<<< HEAD
+* [TutorialsPoint - Intercepting Filter](http://www.tutorialspoint.com/design_pattern/intercepting_filter_pattern.htm)
+* [Presentation Tier Pattern](http://www.javagyan.com/tutorials/corej2eepatterns/presentation-tier-patterns)
+=======
```
Remove this merge marker.

iluwatar commented on README.md:
```diff
@@ -497,9 +515,14 @@ The difference is the intent of the patterns. While Proxy controls access to the
 * [Functional Programming in Java: Harnessing the Power of Java 8 Lambda Expressions](http://www.amazon.com/Functional-Programming-Java-Harnessing-Expressions/dp/1937785467/ref=sr_1_1)
 
 
+>>>>>>> origin/master
```
Remove this merge marker.

iluwatar commented on intercepting-filter/src/main/Client.java:
```diff
@@ -0,0 +1,86 @@
+ * @author joshzambales
+ *
+ */
+public class Client extends JFrame{
```
The fields below have no access specifier e.g. private.

iluwatar commented on intercepting-filter/src/main/Client.java:
```diff
@@ -0,0 +1,86 @@
+		processButton.addActionListener(new ActionListener(){
+			@Override
+			public void actionPerformed(ActionEvent e){
+				jl.setText(sendRequest(jtFields[0].getText()+"&"+jtFields[1].getText()+"&"+jtAreas[0].getText()+"&"+jtFields[2].getText()+"&"+jtAreas[1].getText()));
```
Style issue, I would not use +-operator but [String.format](http://docs.oracle.com/javase/7/docs/api/java/lang/String.html#format(java.lang.String,%20java.lang.Object...)) here

iluwatar commented on intercepting-filter/src/main/FilterChain.java:
```diff
@@ -0,0 +1,48 @@
+ */
+public class FilterChain{
+	private ArrayList<Filter> filters = new ArrayList<Filter>();
+	private Target target;
```
Style issue, make this final since it is initialized in constuctor

iluwatar commented on intercepting-filter/src/main/FilterManager.java:
```diff
@@ -0,0 +1,23 @@
+ *
+ */
+public class FilterManager{
+	FilterChain filterChain;
```
Add access specifier e.g. private

iluwatar commented on intercepting-filter/src/main/Target.java:
```diff
@@ -0,0 +1,58 @@
+ * @author mjoshzambales
+ *
+ */
+public class Target extends JFrame{
```
For the member variables below add access specifier e.g. private

iluwatar commented on intercepting-filter/src/main/Filter.java:
```diff
@@ -0,0 +1,8 @@
+/**
```
I would like to see more documentation. The filters have an important role in the pattern, right?

iluwatar commented on intercepting-filter/src/main/Filter.java:
```diff
@@ -0,0 +1,8 @@
+ * @author joshzambales
+ *
+ */
+public interface Filter{
```
Please add JavaDoc for this method

iluwatar commented on intercepting-filter/src/main/Target.java:
```diff
@@ -0,0 +1,58 @@
+	public void execute(String[] request){
+		dtm.addRow(new Object[]{request[0],request[1],request[2],request[3],request[4]});
+	}
+
```
Add access specifier e.g. private

iluwatar replied:
private class DListener maybe?

iluwatar commented:
Let me know when you've implemented the changes and I'll review again.

joshzambales commented:
I made changes according to the comments:
*Added and revised javadocs as specified
*Added access specifiers
*declared final on target as it is initialized on constructor
*replaced +- operator with String.format
*removed merge markers

iluwatar commented on README.md:
```diff
@@ -497,10 +515,13 @@ The difference is the intent of the patterns. While Proxy controls access to the
 * [Pattern Languages of Program Design](http://www.amazon.com/Pattern-Languages-Program-Design-Coplien/dp/0201607344/ref=sr_1_1)
 * [Martin Fowler - Event Aggregator](http://martinfowler.com/eaaDev/EventAggregator.html)
+* [TutorialsPoint - Intercepting Filter](http://www.tutorialspoint.com/design_pattern/intercepting_filter_pattern.htm)
+* [Presentation Tier Pattern](http://www.javagyan.com/tutorials/corej2eepatterns/presentation-tier-patterns)
```
Should be: Presentation Tier Patterns

iluwatar commented:
Nothing major detected, so I'm off to try to run this on my machine.

iluwatar commented:
Ok, there is one showstopper here. The Java source files have no package declaration (should be com.iluwatar) and they have wrong folder structure.

Put this declaration to each source file: `package com.iluwatar;` Then move all the source files under folder `src/main/java/com/iluwatar`. See the other projects for example.

Let me know when you have done this, and I'll review again.

joshzambales commented:
Fixed folder structure and added package :)

iluwatar commented:
Great! I was able to run it in Eclipse now.

vehpsr commented on intercepting-filter/src/main/java/com/iluwatar/FilterChain.java:
```diff
@@ -0,0 +1,49 @@
+		}else if(tempout[1] == null){
+			return "INVALID Contact Number!";
+		}else if(tempout[0] == null){
+			return "INVALID Name!";
```
Don't know if this will violate pattern structure or filters reusability, but I would like to keep related things together.
For example OrderFilter must know that it works with input array index-field "4", and FilterChain must know that value abscence for tempout[4] is "INVALID ORDER".
I see this magic (numbers and messages) go as constants either to Filter interface, or as public static final fields in each implementation.
```if (tempout[OrderFilter.INDEX] == null) { return OrderFilter.ERROR_MSG; }```

And one step further (although its also can have more problem with pattern consistency):
can we move INDEX and ERROR_MSG to Filter interface as methods, so we can perform polimorfic loop on list of filters instead of if-else chain?

iluwatar commented:
Yes, the magic numbers create an itch for me too. @vehpsr could you please work on this?

vehpsr commented:
I certanly could, but only if @joshzambales will give up on this.
I never heard about this pattern and I never professionally worked with Desctop and Swing, so I may "lead in a wrong direction"...

Another question for me is: would I allow such/additional methods, for example, in chain-of-responsibility pattern implementation?...

iluwatar commented:
Yes, the Chain is certainly similar in implementation. But AFAIK they solve different problems.

There is good discussion about this [here](http://narayan-guttal.blogspot.fi/2014/07/intercepting-filter-vs-chain-of.html) and [here](https://www.linkedin.com/groups/How-is-Intercepting-Filter-pattern-145246.S.75224751). Also, Intercepting Filter is well know and documented in [Core J2EE Patterns](http://www.corej2eepatterns.com/index.htm).

iluwatar commented:
Created issue [#43](https://github.com/iluwatar/java-design-patterns/issues/43). @joshzambales @vehpsr let me know if you want to work on it, otherwise I will take it myself.

iluwatar commented:
@all-contributors please add @joshzambales for code
